openlibrary/core/models.py

```diff
@@ -5,7 +5,7 @@
 import web
 import json
 import requests
-from typing import Any
+from typing import Any, Optional
```

Comment:
Given that we only support Python 3.11, my strong preference would be that we adhere to `ruff rule UP007` except in code that is directly related to Solr.  PEP604 makes code easier to read and eliminates noisy imports.
* https://beta.ruff.rs/docs/rules/non-pep604-annotation

% `ruff rule UP007`
# non-pep604-annotation (UP007)

Derived from the **pyupgrade** linter.

Autofix is sometimes available.

## What it does
Check for type annotations that can be rewritten based on [PEP 604] syntax.

## Why is this bad?
[PEP 604] introduced a new syntax for union type annotations based on the
`|` operator. This syntax is more concise and readable than the previous
`typing.Union` and `typing.Optional` syntaxes.

This rule is enabled when targeting Python 3.10 or later (see:
[`target-version`]). By default, it's _also_ enabled for earlier Python
versions if `from __future__ import annotations` is present, as
`__future__` annotations are not evaluated at runtime. If your code relies
on runtime type annotations (either directly or via a library like
Pydantic), you can disable this behavior for Python versions prior to 3.10
by setting [`pyupgrade.keep-runtime-typing`] to `true`.

## Example
```python
from typing import Union

foo: Union[int, str] = 1
```

Use instead:
```python
foo: int | str = 1
```

## Options
- `target-version`
- `pyupgrade.keep-runtime-typing`

[PEP 604]: https://peps.python.org/pep-0604/

Reply:
@cclauss why don't we want this rule for solr?
It seems the codebase is kinda split about as many uses of `Optional` as there are `| None` 

I also see you turned off this rule in our config, is that just because of solr?
https://github.com/internetarchive/openlibrary/blob/c7c1d52c395ced1dcb0e98eb5409ac4d85d2f137/pyproject.toml#L70

Reply:
`openlibrary/solr/types_generator.py` I believe still requires pre-PEP604 types.

Reply:
I believe this was due to an issue with the FnToCLI ? But I updated that a while back ` | None` works. So whichever!